### PR TITLE
Autocomplete: add analytics event agent tests

### PR DESCRIPTION
**Pull request by valerybugakov**
Follow-up addition with agent tests for https://github.com/sourcegraph/cody/pull/4984. No behaviour changes

## Test plan

New agent tests + CI

### CODE REVIEW

File: agent/src/autocomplete.test.ts [hunk 1]
```diff
@@ -31,6 +33,30 @@
             position: { line: 1, character: 4 },
             triggerKind: 'Invoke',
         })
+        const completionID = completions.items[0].id
+        await client.notify('autocomplete/completionSuggested', { completionID })
+        await sleep(400)
```
Comment (RXminuS) [inline agent/src/autocomplete.test.ts:38]: We shouldn't actually sleep here. There's timer helpers to advance time

File: agent/src/autocomplete.test.ts [hunk 2]
```diff
@@ -40,20 +66,31 @@
...
         const completions = await client.request('autocomplete/execute', {
             uri: uri.toString(),
             position: { line: 1, character: 4 },
             triggerKind: 'Invoke',
         })
+        const completionID = completions.items[0].id
         const texts = completions.items.map(item => item.insertText)
+
+        await client.notify('autocomplete/completionSuggested', { completionID })
+        await sleep(800) // Wait for the completion visibility timeout (750ms) to elapse
```
Comment (RXminuS) [inline agent/src/autocomplete.test.ts:84]: Same here https://vitest.dev/api/vi.html#vi-usefaketimers

File: agent/bindings/kotlin/lib/src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/Constants.kt [hunk 3]
```diff
@@ -110,6 +110,8 @@ object Constants {
   const val IDEEXTENSION = "IDEEXTENSION"
   const val ignore = "ignore"
   const val use = "use"
+  const val `tree-sitter` = "tree-sitter"
```
Comment (RXminuS) [inline agent/bindings/kotlin/lib/src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/Constants.kt:113]: How is this the only diff in a generated file? That seems suspicious

**Review by RXminuS — CHANGES_REQUESTED** [review]
Mainly the timers

File: agent/bindings/kotlin/lib/src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/CodyAgentServer.kt [hunk 4]
```diff
@@ -112,6 +112,8 @@ interface CodyAgentServer {
   fun testing_progressCancelation(params: Testing_ProgressCancelationParams): CompletableFuture<Testing_ProgressCancelationResult>
   @JsonRequest("testing/reset")
   fun testing_reset(params: Null?): CompletableFuture<Null?>
+  @JsonRequest("testing/autocomplete/completionEvent")
+  fun testing_autocomplete_completionEvent(params: CompletionItemParams): CompletableFuture<CompletionBookkeepingEvent?>
```
Comment (RXminuS) [inline agent/bindings/kotlin/lib/src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/CodyAgentServer.kt:116]: Is this needed? Isn't the completion event sent so you can just listen for it?

File: agent/src/autocomplete.test.ts (hunk 2 above)
Comment (valerybugakov) [reply agent/src/autocomplete.test.ts:84]: @RXminuS, unfortunately, fake timers don't work across node processes (let me know if you know a way of using them in agent tests; maybe I'm missing something). Several months ago, I started looking into adding support for it but didn't have an opportunity to prioritize this improvement. 

Another process controls the timer I need to advance, so `vitest` helpers do not affect it in a desired way.

File: agent/bindings/kotlin/lib/src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/Constants.kt (hunk 3 above)
Comment (valerybugakov) [reply agent/bindings/kotlin/lib/src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/Constants.kt:113]: This is produced by `pnpm generate-agent-kotlin-bindings`. No manual changes from my side.

File: agent/bindings/kotlin/lib/src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/CodyAgentServer.kt (hunk 4 above)
Comment (valerybugakov) [reply agent/bindings/kotlin/lib/src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/CodyAgentServer.kt:116]: Without this testing hook, there's no way to get the latest completion event.

File: agent/src/autocomplete.test.ts (hunk 2 above)
Comment (RXminuS) [reply agent/src/autocomplete.test.ts:82]: Ah sorry, forgot this was an agent test. I won't let it block then.

We could fairly easily implement it as a testing message or we can start the process with the inspector enabled and we can just advance it externally. Maybe at least open an issue for it so it doesn't get lost?

And maybe add a @slowTimer tag or something on this test title so we can easily find them later?

**Review by RXminuS — APPROVED** [review]

File: agent/src/autocomplete.test.ts (hunk 2 above)
Comment (valerybugakov) [reply agent/src/autocomplete.test.ts:84]: Added TODO with a comment about this 👍

**Pull request merged**